dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,698 @@
+    }
+
+    def to_db_decimal_type(
+        self, precision: Optional[int], scale: Optional[int]
+    ) -> pa.Decimal128Type:
+        precision, scale = self.decimal_precision(precision, scale)
+        return pa.decimal128(precision, scale)
+
+    def to_db_datetime_type(
+        self, precision: Optional[int], table_format: TTableFormat = None
+    ) -> pa.TimestampType:
+        unit: str = TIMESTAMP_PRECISION_TO_UNIT[self.capabilities.timestamp_precision]
+        return pa.timestamp(unit, "UTC")
+
+    def to_db_time_type(
+        self, precision: Optional[int], table_format: TTableFormat = None
+    ) -> pa.Time64Type:
+        unit: str = TIMESTAMP_PRECISION_TO_UNIT[self.capabilities.timestamp_precision]
+        return pa.time64(unit)
+
+    def from_db_type(
+        self,
+        db_type: pa.DataType,
+        precision: Optional[int] = None,
+        scale: Optional[int] = None,
+    ) -> TColumnType:
+        if isinstance(db_type, pa.TimestampType):
+            return dict(
+                data_type="timestamp",
+                precision=UNIT_TO_TIMESTAMP_PRECISION[db_type.unit],
+                scale=scale,
+            )
+        if isinstance(db_type, pa.Time64Type):
+            return dict(
+                data_type="time",
+                precision=UNIT_TO_TIMESTAMP_PRECISION[db_type.unit],
+                scale=scale,
+            )
+        if isinstance(db_type, pa.Decimal128Type):
+            precision, scale = db_type.precision, db_type.scale
+            if (precision, scale) == self.capabilities.wei_precision:
+                return cast(TColumnType, dict(data_type="wei"))
+            return dict(data_type="decimal", precision=precision, scale=scale)
+        return cast(
+            TColumnType,
+            without_none(
+                dict(
+                    data_type=self.dbt_to_sct.get(db_type, "text"),
+                    precision=precision,
+                    scale=scale,
+                )
+            ),
+        )
+
+
+@lancedb_batch_error
+def upload_batch(
+    records: List[DictStrAny],
+    /,
+    *,
+    db_client: DBConnection,
+    table_name: str,
+    write_disposition: TWriteDisposition,
+    id_field_name: Optional[str] = None,
+) -> None:
+    """Inserts records into a LanceDB table with automatic embedding computation.
+
+    Args:
+        records: The data to be inserted as payload.
+        db_client: The LanceDB client connection.
+        table_name: The name of the table to insert into.
+        id_field_name: The name of the ID field for update/merge operations.
+        write_disposition: The write disposition - one of 'skip', 'append', 'replace', 'merge'.
+
+    Raises:
+        ValueError: If the write disposition is unsupported, or `id_field_name` is not
+            provided for update/merge operations.
+    """
+
+    try:
+        tbl = db_client.open_table(table_name)
+    except FileNotFoundError as e:
+        raise LanceDBBatchError(e) from e
```

Comment:
I'm assuming this is done like this because the filenotfounderror means different things in different lancedb contexts right? You could directly raise the correct dlt error here.

Reply:
I removed `LanceDBBatchError` class and directly raised the appropriate dlt destination errors as requested